### PR TITLE
Editable author on model page

### PR DESCRIPTION
**Pull request by joeyballentine**
Only issue is that it doesn't allow you to add or remove authors, if there are multiple. I don't see this as a huge issue at the moment though.

### CODE REVIEW

File: src/pages/models/[id].tsx [hunk 1]
```diff
@@ -42,12 +43,12 @@ const dummyImages = [
...
 ];
 
-export default function Page({ modelId, modelData }: Props) {
+export default function Page({ modelId, modelData, userIds }: Props) {
     const { webApi, editMode } = useWebApi();
     const model = useCurrent(webApi, 'model', modelId, modelData);
 
     const updateModelProperty = useCallback(
-        (key: string, value: string) => {
+        (key: string, value: string | string[] | UserId | UserId[]) => {
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:51]: Let's make this more robust.

```suggestion
        <K extends keyof Model>(key: K, value: Model[K]) => {
```

File: src/pages/models/[id].tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ interface Params extends ParsedUrlQuery {
 interface Props {
     modelId: ModelId;
     modelData: Model;
+    userIds: UserId[];
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:21]: I don't think that we should include data that is only used in edit mode in the static page data. This just needlessly bloats the page size for normal viewers.

I think we should use the webApi to fetch data needed for edit mode.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: src/pages/models/[id].tsx (hunk 2 above)
Comment (joeyballentine) [reply src/pages/models/[id].tsx:21]: Makes sense.

File: src/pages/models/[id].tsx [hunk 3]
```diff
@@ -47,13 +47,21 @@ export default function Page({ modelId, modelData }: Props) {
...
         [webApi, modelId, model]
     );
 
+    const [userIds, setUserIds] = useState<UserId[]>([]);
+    useEffect(() => {
+        webApi?.users
+            .getIds()
+            .then((data) => setUserIds(data))
+            .catch((e) => console.error(e));
+    }, [webApi?.users]);
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:63]:
```suggestion
    const { userData } = useUsers();
```

File: src/pages/models/[id].tsx [hunk 4]
```diff
@@ -91,11 +99,40 @@ export default function Page({ modelId, modelData }: Props) {
...
+                                                        }}
+                                                    >
+                                                        {userIds.map((user) => (
+                                                            <option
+                                                                key={user}
+                                                                value={user}
+                                                            >
+                                                                {user}
+                                                            </option>
+                                                        ))}
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:126]:
```suggestion
                                                        {[...userData].map(([userId, user]) => (
                                                            <option
                                                                key={userId}
                                                                value={userId}
                                                            >
                                                                {user.name}
                                                            </option>
                                                        ))}
```

File: src/pages/models/[id].tsx [hunk 5]
```diff
@@ -91,11 +99,40 @@ export default function Page({ modelId, modelData }: Props) {
...
+                                                    </select>
+                                                );
+                                            }
+                                            return (
+                                                <React.Fragment key={userId}>
+                                                    <Link href={`/users/${userId}`}>{userId}</Link>
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:132]:
```suggestion
                                                    <Link href={`/users/${userId}`}>{userData.get(userId) ?? `unknow user:${userId}`}</Link>
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: src/pages/models/[id].tsx (hunk 3 above)
Comment (joeyballentine) [reply src/pages/models/[id].tsx:63]: Oh, lol

**Comment by joeyballentine** [comment]
@RunDevelopment can you lint this for me?

**Comment by RunDevelopment** [comment]
Sure!

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**